src/utils/getLanguageSlug.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+const config = require("../_data/config.json");
+
+module.exports = (lang) => {
+    /* Trim the country code from the end of the language. */
+    const shortLang = lang.split("-")[0];
+
+    /* Create array of all supported language codes with the country code trimmed. */
+    const codes = config.languages.reduce((accumulator, currentValue) => {
+        accumulator.push(currentValue.code.split("-")[0]);
+        return accumulator;
+    }, []);
```

Comment:
I think what you are doing here is to take an array like 

```javascript
[{code: "en-CA"}, {code: "en-GB"}, {code: "en-US"}, {code: "fr-CA"}, {code: "fr-FR"}, {code: "es-MX"}]` 
```

and convert it to an array like:

```javascript
["en", "en", "en", "fr", "fr", "es"]
```

If so, you could simply the code by using [Array.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) instead. However, you could continue to use reduce and instead of outputting an array, output an Object of counts. That is with the key the trimmed lang and the value the number of occurrences. Such as:

```json
{
    "en": 3,
    "fr": 2,
    "es" 1
}
```

Something like the following should accomplish that. 

```javascript
const codes = config.languages.reduce((accumulator, currentValue) => {
    let lang = currentValue.code.split("-")[0];
    accumulator[lang] = (++accumulator[lang] || 1);
    return accumulator;
}, {});
```

Reply:
I like this alternate usage of the reducer— I'm going to try it.

Reply:
A bit sleazy having the conditional double side-effect there - one might want to say
````
accumulator[lang] = (accumulator[lang] || 0) + 1; 
````
to be a touch clearer - note that in the bad old days (C++) such expressions were actually undefined!
But it's not really clear why the counts are actually required - isn't it enough to simply say
````
accumulator[lang] = true
````
and then emit an Object.keys() of the result?

Note that the original method is also fine, after passing the result through a Set comprehension as in this answer: https://stackoverflow.com/a/33121880

Reply:
> But it's not really clear why the counts are actually required - isn't it enough to simply say
> 
> ```
> accumulator[lang] = true
> ```
> 
> and then emit an Object.keys() of the result?

Not really! The intent here is to find languages with more than one national/regional variant so that slugs don't overwrite each other.

For example, let's say we have a site with `en-CA` and `fr-CA` as in the example config. It's cleaner to have URLs constructed like this: `/en/something` or `/fr/quelque-chose` so we want to trim the country code off the language.

However, imagine a more complex Trivet-based site which has many variations: `es-ES`, `es-MX`, `pt-BR`, `pt-PT`. If we were to follow the same logic and trim the country code that would break the site because both the Spanish (Spain) and Spanish (Mexico) languages would end up with the `/es` language slug.

So by building an object of short language slugs ([ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes)) with the count for each, we can use the short code if the count is 1 but use the long codes for languages with a count > 2. Does this make sense, @amb26? I will make the change you suggest related to the conditional double side-effect.

Reply:
@amb26 I think the issue is that you need to know when there are more than 1. If we just added a boolean flag or used a set, we'd know that there was an occurrence of a language, but not the number of locales present. The number of locales is used to determine if the url can be simplified to something like `/fr/post/` or requires the longer form like `/fr-CA/posts/` and `/fr-FR/posts/`.